pkg/influx/api.go

```diff
@@ -51,30 +52,45 @@ func NewAPI(logger log.Logger, cfg APIConfig) (*API, error) {
 		return nil, err
 	}
 
+	level.Info(logger).Log("msg", "Creating GRPC connection to", "addr", cfg.DistributorEndpoint)
 	conn, err := grpc.Dial(cfg.DistributorEndpoint, dialOpts...)
 	if err != nil {
+		level.Error(logger).Log("msg", "Failed to connect to server", "err", err)
 		return nil, err
 	}
 
 	distClient := distributorpb.NewDistributorClient(conn)
 
+	recorder := NewRecorder(prometheus.NewRegistry())
+
 	return &API{
-		logger: logger,
-		client: distClient,
+		logger:   logger,
+		client:   distClient,
+		recorder: recorder,
 	}, nil
 }
 
 // HandlerForInfluxLine is a http.Handler which accepts Influx Line protocol and converts it to WriteRequests.
 func (a *API) handleSeriesPush(w http.ResponseWriter, r *http.Request) {
 	maxSize := 100 << 10 // TODO: Make this a CLI flag. 100KB for now.
 
+	//userID, err := user.ExtractOrgID(r.Context())
+	//if err != nil {
+	//	level.Error(a.logger).Log("msg", "error extracting org ID", "err", err)
+	//	panic(err)
+	//}
+
+	beforeConversion := time.Now()
+
 	ts, err := parseInfluxLineReader(r.Context(), r, maxSize)
 	if err != nil {
-		fmt.Println("error decoding line protocol data", err)
+		a.recorder.measureMetricsRejected(len(ts))
+		level.Error(a.logger).Log("msg", "error decoding line protocol data", "err", err)
```

Comment:
error level logging is only for internal issues that are not recoverable by the server. Warning is for server problems that are not desirable, but are fine. User errors like this are perfectly expected, so this should just be Info.